x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -136,16 +138,19 @@ function transformResponseToServiceAnomalies(
 }
 
 export async function getMLJobIds(
-  ml: Required<Setup>['ml'],
+  ml: {
+    anomalyDetectors: ReturnType<MlPluginSetup['anomalyDetectorsProvider']>;
+  },
```

Comment:
Perhaps just change the signature:


```suggestion
  anomalyDetectors: Required<Setup>['ml']['anomalyDetectors'],
```